src/playercards/cards/GabrielCarillo1.ttslua

```diff
@@ -0,0 +1,79 @@
+require("playercards/CardsWithHelper")
+local blessCurseManagerApi = require("chaosbag/BlessCurseManagerApi")
+local playermatApi         = require("playermat/PlayermatApi")
+
+-- intentionally global
+hasXML                     = true
+isHelperEnabled            = false
+local updated, loopId, tokenAdded
+
+function updateSave()
+  self.script_state = JSON.encode({
+    isHelperEnabled = isHelperEnabled,
+    loopId = loopId
+  })
+end
+
+function onLoad(savedData)
+  if savedData and savedData ~= "" then
+    local loadedData = JSON.decode(savedData)
+    isHelperEnabled = loadedData.isHelperEnabled
+    loopId = loadedData.loopId
+  end
+  if isHelperEnabled then updateDisplay() end
+end
+
+function shutOff()
+  if loopId then
+    Wait.stop(loopId)
+    loopId = nil
+  end
+end
+
+function initialize()
+  maybeUpdateButtonState()
+  loopId = Wait.time(maybeUpdateButtonState, 1, -1)
+end
+
+function resolveToken(player, _, tokenType)
+  if not updated then return end
+
+  local matColor
+  if player.color == "Black" then
+    matColor = playermatApi.getMatColorByPosition(self.getPosition())
+  else
+    matColor = playermatApi.getMatColor(player.color)
+  end
+
+  blessCurseManagerApi.addToken("Curse", player.color)
+  tokenAdded = true
```

Comment:
Considering that there are ways to ready cards outside of upkeep, I would not track whether the ability was used already